source/custom/visa_service.custom.cpp

```diff
@@ -131,6 +185,30 @@ ::grpc::Status VisaService::ConvertApiErrorStatusForViObject(::grpc::ServerConte
 }
 
 
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+::grpc::Status VisaService::CloseEvent(::grpc::ServerContext* context, const CloseEventRequest* request, CloseEventResponse* response)
+{
+  if (context->IsCancelled()) {
+    return ::grpc::Status::CANCELLED;
+  }
+  try {
+    ViEvent event_handle = request->event_handle();
+    auto status = library_->CloseEvent(event_handle);
+    if (!status_ok(status)) {
+      return ConvertApiErrorStatusForViSession(context, status, 0);
+    }
+    auto matching_event_lambda = [&](const VisaAsyncOperation& v) { return v.event == event_handle; };
```

Comment:
Consider: could events be implemented as a grpc-device session? 2 possible benefits:

1. Allows for "client allocated IDs" via "named sessions". This enables replay APIs because you can have the same named event across sessions.
1. If it works, the session repository could be helpful reusable code.

I think xnet does something like this with the dbrefs.

I realize this is not a minor suggestion. And I think the way the VISA API works, there's a lot of low level back and forth (like the memory API) that wouldn't work for replay testing anyway. Just an idea; in case it's useful.

Reply:
This was brought up earlier in the development and I declined for these reasons:
- Events happen when the driver detects them. 
- You cannot just access one when you want.
- They are extremely temporal / short-lived.

Reply:
@astarche, here's the previous discussion thread: https://github.com/ni/grpc-device/pull/992#discussion_r1313515516